R/CPOAuxiliary.R

```diff
@@ -0,0 +1,373 @@
+#' @title CPO: Composable Preprocessing Operators
+#'
+#' @description
+#' FIXME to come
+#'
+#' @family CPO
+#' @name CPO
+NULL
+
+
+### Generics
+
+#' @title CPO Composition / Attachment operator
+#'
+#' @description
+#' This operator \dQuote{pipes} data from the source into the target object.
+#'
+#' If both objects are a \link{CPO} object, they will be composed. A new object,
+#' representing the operation of performing both object's operations in succession,
+#' will be created, which can be handled like a new \link{CPO} object.
+#'
+#' If the left object is a \code{data.frame} or a \code{link{Task}}, the
+#' transformation operation will be applied to this data, and the same resulting
+#' data will be returned.
+#'
+#' If the right object is a \code{\link{Learner}}, the CPO will be attached to
+#' this learner. The same operation will be performed during the \dQuote{train} and
+#' \dQuote{predict} phase; the behaviour during the predict phase may furthermore
+#' be depend on the training data.
+#'
+#' Note that you can not link a \code{data.frame} or \code{\link{Task}} directly
+#' to a \code{\link{Learner}}, since this operation is not algebraically associative
+#' with the composition of CPOs. Use \code{\link{train}} for this.
+#'
+#' @param cpo1 [\code{data.frame} | \code{\link{Task}} | \code{\link{CPO}}]\cr
+#'   The source object.
+#' @param cpo2 [\code{\link{CPO}} | \code{\link{Learner}}]\cr
+#'   The target object.
+#'
+#' @family CPO
+#' @examples
+#' # PCA-rotate pid.task
+#' rotated.pid.task = pid.task %>>% cpoPca()
+#'
+#' # Centering / Scaling *after* PCA
+#' neoPCA = cpoPca(center = FALSE, scale = FALSE, id = "pca") %>>% cpoScale()
+#'
+#' # Attach the above to learner
+#' pcaLogreg = neoPCA %>>% makeLearner("classif.logreg")
+#'
+#' @export
+`%>>%` = function(cpo1, cpo2) {
+  UseMethod("%>>%")
+}
+
+#' @export
+`%>>%.default` = function(cpo1, cpo2) {
+  stopf("%%>>%% not defined for objects of class c(%s)", paste0('"', class(cpo1), '"', collapse = ", "))
+}
+
+#' @export
+`%>>%.data.frame` = function(cpo1, cpo2) {
+  name = deparse(substitute(cpo1), 20)[1]
```

Comment:
I just want a short identifying "name" for the task. The task will be destroyed after application of `%>>%` anyways, so the user only sees this if there is some error message.